src/tbb/arena_slot.cpp

```diff
@@ -142,12 +147,22 @@ d1::task* arena_slot::get_task(execution_data_ext& ed, isolation_type isolation)
 
     __TBB_ASSERT( (std::intptr_t)tail.load(std::memory_order_relaxed) >= 0, nullptr );
     __TBB_ASSERT( result || tasks_omitted || is_quiescent_local_task_pool_reset(), nullptr );
+
+    if (result) {
```

Comment:
To compiler optimization when `__TBB_STATISTICS==0`:
```suggestion
    if (__TBB_STATISTICS && result) {
```